content/articles/php-sessions-and-cookies-to-create-a-secure-remember-me-section-in-a-login-script/Index.md

```diff
@@ -0,0 +1,356 @@
+### Introduction
+In the Login Script, the 'Remember Me' functionality allows users who have logged in to keep their status. A user's logged-in status is serialized in PHP `sessions`, `cookies`, or other similar storage when they select the `Remember Me` option.
+In this edge, you will learn how to create a persistent PHP web application.
+#### Table of contents
+- [Introduction](#introduction)
+- [Preserve logged in state](#create-cookies-to-preserve-the-logged-in-state)
+- [validate remembered login](#validate-remembered-login-with-php-session-and-cookies)
+- [Clear remembered login](#clear-remembered-login-with-session-and-cookies-on-logout)
+- [Controller classes](#auth-and-dbcontroller-classes)
+- [Conclusion](#conclusion)
+- [check the code snippets at github](https://github.com/Juskam/secure-remember-me-for-login-using-php-session-and-cookies)
+Make sure to look at the source code on GitHub so you know what files to create.
+### Create cookies to preserve the logged-in state
+To obtain the `username` and `password`, I've created a login form. This form has a 'Remember Me' checkbox for the user's convenience, which allows the user to keep his logged-in state. When a user provides their login information, PHP receives it and compares it to the database of members.
+On successful login, if the user selected ‘Remember Me’ then the logged-in status is stored in PHP `session` and `cookies`.
+Because storing the plain password in the cookie is a security flaw, the authentication keys are produced using random integers. These keys are hashed and saved in a database with a one-month expiration time. When the timer runs out, the expiration flag is set to `0` and the keys are turned off.

```

Comment:
Is the expiry restricted to one month?

Reply:
@jumamiller, from this tutorial I intended to keep users logged in for one month and that explains the restriction to one month